django/contrib/admin/templatetags/admin_list.py

```diff
@@ -224,8 +239,7 @@ def link_in_col(is_first, field_name, cl):
             if f is None or f.auto_created:
                 if field_name == "action_checkbox":
                     row_classes = ["action-checkbox"]
-                boolean = getattr(attr, "boolean", False)
-                result_repr = display_for_value(value, empty_value_display, boolean)
+                result_repr = get_result_repr(attr, value, empty_value_display)
```

Comment:
I'm not sure why you want to add two nested helpers here :thinking: Do we need them? It should be enough to check `boolean` for a property, e.g.
```diff
diff --git a/django/contrib/admin/templatetags/admin_list.py b/django/contrib/admin/templatetags/admin_list.py
index 9bff84d501..498e27e77d 100644
--- a/django/contrib/admin/templatetags/admin_list.py
+++ b/django/contrib/admin/templatetags/admin_list.py
@@ -225,6 +225,9 @@ def items_for_result(cl, result, form):
                 if field_name == "action_checkbox":
                     row_classes = ["action-checkbox"]
                 boolean = getattr(attr, "boolean", False)
+                # Set boolean for attr that is a property, if defined.
+                if isinstance(attr, property) and hasattr(attr, 'fget'):
+                    boolean = getattr(attr.fget, 'boolean', False)
                 result_repr = display_for_value(value, empty_value_display, boolean)
                 if isinstance(value, (datetime.date, datetime.time)):
                     row_classes.append("nowrap")
```

Reply:
Yeah right! The two nested helpers were making it overly complex. Thanks for pointing this out.